### DISCUSSION TITLE
Get file zip from zip archive ?

### DISCUSSION THREAD

**Discussion by CouinCouin** [wqweto/ZipArchive#23] [General]
Hi (wqweto), 

Using this model : 
```
.OpenArchive "C:\TEMP\aaa.zip"
.Extract baOutput, "mem/report.pdf"
WriteBinaryFile "C:\TEMP\report.pdf", baOutput
```
Is there a way to know the size of "mem/report.pdf" before writing it in "C:\TEMP\" ?

Tried by search in the doc but "size" word appears too much times to makes searching revelant.

Thanks :)

**Comment by wqweto** [comment]
Here is a snippet which gets output size before decompressing the file in memory
```vba
    Dim lIdx        As Long
    Dim vInfo       As Variant
    Dim lSize       As Long
    Dim baOutput()  As Byte
    
    With New cZipArchive
        .OpenArchive "D:\TEMP\aaa.zip"
        For lIdx = 0 To .FileCount - 1
            vInfo = .FileInfo(lIdx)
            If LCase$(vInfo(0)) = "mem\report.pdf" Then
                lSize = vInfo(3)
                Debug.Print "Reported output size is " & lSize
                Exit For
            End If
        Next
        .Extract baOutput, "mem\report.pdf"
        Debug.Print "Decompressed size is " & UBound(baOutput) + 1
    End With
    WriteBinaryFile "D:\Temp\report.pdf", baOutput
```
You can check out [`cVbZip.cls`](https://github.com/wqweto/ZipArchive/blob/master/test/vbzip/cVbZip.cls) sample of a zip utility which includes almost everything which can be done with the class incl. listing content of the archive with filenames, compressed/decompressed sizes, etc.

Here is the enum with indexes into `vInfo` above as used by `cVbZip` class
```vba
Private Enum ZipInfoIndexesEnum
    zipIdxFileName
    zipIdxAttributes
    zipIdxCrc32
    zipIdxSize
    zipIdxCompressedSize
    zipIdxComment
    zipIdxLastModified
    zipIdxMethod
    zipIdxOffset
    zipIdxFlags
End Enum
```
You can use these like `vInfo(zipIdxFileName)` instead of `vInfo(0)` for filename and `vInfo(zipIdxSize)` instead of  `vInfo(3)` for size.

**Comment by CouinCouin** [comment]
Hi , Perfect :) 
Thanks !

**Comment by wqweto** [comment]
Btw, in the latest version of the class one can get entry's uncompressed file size using `FileInfo` property with original filename like this
```
    With New cZipArchive
        .OpenArchive "D:\TEMP\aaa.zip"
        Debug.Print .FileInfo("report.pdf", zipIdxSize)
    End With
```
